src/opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -556,6 +556,10 @@ EclipseGrid::EclipseGrid(const Deck& deck, const int * actnum)
         return m_pinchMaxEmptyGap;
     }
 
+    bool EclipseGrid::maxEmptyGapDefaulted() const {
+        return m_pinchMaxEmptyGap ==
+            ParserKeywords::PINCH::MAX_EMPTY_GAP::defaultValue;
```

Comment:
Does this hold for `FIELD` or `LAB` units?  The `defaultValue` is always `METRIC`, so if `m_pinchMaxEmptyGap` is SI, then that comparison won't be true.  In that case we might want to make `m_pinchMaxEmptyGap` into an `optional<double>` and have `nullopt` represent "this item is defaulted".

Reply:
there you go, now we get bitten by a [discussion about principles](https://github.com/OPM/opm-common/pull/3456#discussion_r1151800152) with no real value. Seems like simple is always better.

I am really tempted to go back to my initial code...

Reply:
Any nice catch. I need to find a solution.